openapi3/refs_test.go

```diff
@@ -332,3 +334,65 @@ components:
 	_, _, err = ptr.Get(doc)
 	require.Error(t, err)
 }
+
+func TestSchemaRef_Extensions(t *testing.T) {
+	spec := []byte(`
+openapi: 3.0.0
+components:
+  schemas:
+    Pet:
+      x-order: "pet"
+      type: object
+      required:
+        - id
+        - name
+      properties:
+        id:
+          type: integer
+          format: int64
+    PetsXOrder:
+      type: array
+      items:
+        $ref: '#/components/schemas/Pet'
+        x-order: "petsXOrder"
+
+    PetsName:
+      type: array
+      items:
+        $ref: '#/components/schemas/Pet'
+        name: 1
+`[1:])
+
+	loader := NewLoader()
+	doc, err := loader.LoadFromData(spec)
+	assert.NoError(t, err)
+
+	// captures extension
+	ref := doc.Components.Schemas["PetsXOrder"].Value.Items
+	assert.Equal(t, "petsXOrder", ref.Extensions["x-order"])
+	assert.Equal(t, "pet", ref.Value.Extensions["x-order"])
+
+	err = ref.Validate(context.Background())
+	require.EqualError(t, err, `extra sibling fields: [x-order]`)
+
+	err = ref.Validate(context.Background(), AllowExtraSiblingFields("x-order"))
```

Comment:
An other `func Allow/DisallowExtensionsInRefPosition() ValidationOption { .. }` would be nice (names suggestions welcomed)